### PR TITLE
Update documentation and coreJSON submodule

### PR DESCRIPTION
**Pull request by divekarshubham**
*Issue #, if available:*

*Description of changes:*

* Updated README with the latest references and some refactoring
* Updating coreJSON to v3.0.2
* Renamed `size_table.html` to `.md`
* Updated changelog to show August release

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,13 +57,12 @@ The following table lists equivalent API function signatures in v2.0.0 (Release
...
+
+Refer to [OTA Migration document](https://docs.aws.amazon.com/freertos/latest/portingguide/porting-migration-ota.html) for the summary of updates to the API.
+[Migration document for OTA PAL](https://docs.aws.amazon.com/freertos/latest/portingguide/porting-migration-ota-pal.html) also provides a summary of updates required for upgrading the OTA-PAL to work with v3.0.0 of the library.
 
-| Platform | Location |
-| :-: | :-: |
-| POSIX | [AWS IoT Device SDK for Embedded C](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/ota) |
 ## Porting
 
 In order to support AWS IoT Over-the-air Updates on your device, it is necessary to provide the following components:
```
Comment (pvyawaha) [inline README.md:68]: MQTT is required but HTTP is optional and only required when HTTP is used for file download

Comment (divekarshubham) [reply README.md:68]: Good catch, updated.

Comment (pvyawaha) [reply README.md:68]: Thanks , but can we clarify that when MQTT is used for download then only MQTT interface is required , but for HTTP download both MQTT and HTTP is required as MQTT in this case is used for job notifications and status updates whereas HTTP is for file download

Comment (divekarshubham) [reply README.md:68]: Done

**Review by pvyawaha — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -73,9 +72,34 @@ In order to support AWS IoT Over-the-air Updates on your device, it is necessary
 
 1. [MQTT Interface](https://docs.aws.amazon.com/embedded-csdk/202103.00/lib-ref/libraries/aws/ota-for-aws-iot-embedded-sdk/docs/doxygen/output/html/ota_porting.html#ota_porting_mqtt)
 
+For enabling data transfer over HTTP dataplane the following component should also be provided:
+
 1. [HTTP Interface](https://docs.aws.amazon.com/embedded-csdk/202103.00/lib-ref/libraries/aws/ota-for-aws-iot-embedded-sdk/docs/doxygen/output/html/ota_porting.html#ota_porting_http)
 
-## Generating documentation
+**NOTE** When using OTA over HTTP dataplane, MQTT is required for control plane operations and should also be ported.
```
Comment (aggarg) [inline README.md:79]:
```suggestion
**NOTE** When using OTA over HTTP dataplane, MQTT is required for control plane operations and should also be provided.
```

File: README.md [hunk 3]
```diff
@@ -73,9 +72,34 @@ In order to support AWS IoT Over-the-air Updates on your device, it is necessary
...
+## Reference examples
+
+Please refer to the demos of the AWS IoT Over-the-air Updates library in the following location for a reference example on POSIX:
```
Comment (aggarg) [inline README.md:82]:
```suggestion
Please refer to the demos of the AWS IoT Over-the-air Updates library in the following location for reference examples on POSIX and FreeRTOS:
```

**Review by paulbartell — APPROVED** [review]

**Review by pvyawaha — APPROVED** [review]

**Pull request merged**